### PR TITLE
Small prompt shifts. Changing some names, etc. (PR 2)

### PR DESCRIPTION
**Pull request by knod**
Some of these are longer and a bit more awkward, but I think it's easier to tell what they're doing.

Going to change `OnLeavePrompt` to `FeedbackPrompt` next. Figured it might be too confusing to do all in one step considering the previous `FeedbackPrompt` component.

Also, may need to move `FeedbackForm` out of prompts. Not exactly sure what the definition of a 'prompt' is.

### CODE REVIEW

File: src/components/prompts/ReactRouterConfirmLeave.js [hunk 1]
```diff
@@ -2,26 +2,26 @@ import React from 'react';
 import { Prompt } from 'react-router-dom';
 
 /**
- * Customize React Router on leave prompt to use <OnLeavePrompt>
+ * Customize React Router on leave/unmount prompt to use <OnLeavePrompt>
```
Comment (wacii) [inline src/components/prompts/ReactRouterConfirmLeave.js]: Not sure what `<ReactRouter.Prompt>` has to do with unmounting.

File: src/components/prompts/ReactRouterConfirmLeave.js [hunk 2]
```diff
@@ -2,26 +2,26 @@ import React from 'react';
...
  * Hijacks history to show a modal first before
  * letting history take its course.
  * 
  * @param props {object}
- * @param props.prompt {function}
+ * @param props.askForFeedback {function}
  * @param props.message {string} - Passed indirectly to <OnLeavePrompt>
  * @param props.confirmer {object} - For hijacking history
- * @param props.isBlocking {boolean} - Whether the prompt should be shown
+ * @param props.isBlocking {boolean} - Whether this prompt should be shown
```
Comment (wacii) [inline src/components/prompts/ReactRouterConfirmLeave.js]: This component doesn't return any visible component. It just sets up a callback. But maybe it is calling the component a prompt in the first place which is the problem. Or maybe I'm just being too picky.

File: src/containers/VisitPage.js [hunk 3]
```diff
@@ -107,50 +108,63 @@ class VisitPage extends Component {
     this.setState({ client: nextClient });
   };
 
-  resetClient = () => {
+  resetClient = (shouldReset) => {
+
+    if (!shouldReset) {
+      return;
+    }
```
Comment (wacii) [inline src/containers/VisitPage.js]: Calling `resetClient()` when you don't actually want to reset the client is a little weird. Not a big deal though.

File: src/components/prompts/ResetAnytime.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ const ResetAnytime = function (props) {
   return (
     <div>
       <Button
-        onClick={ props.resetClient }
+        onClick={ props.askToResetClient }
```
Comment (wacii) [inline src/components/prompts/ResetAnytime.js]: This is certainly a better name for it.

File: src/components/prompts/ErrorPrompt.js [hunk 5]
```diff
@@ -5,14 +5,15 @@ import React from 'react';
  * @extends React.Component
  *
  * @param props {object}
- * @param props.callback {onLeaveCallback}
+ * @param props.callback {function}
```
[the remark this answers is not in the document]
Comment (knod) [reply src/components/prompts/ErrorPrompt.js]: The callback is still used, though, right? Do you mean that documentation should be added to describe it?

File: src/components/prompts/ErrorPrompt.js [hunk 6]
```diff
@@ -5,14 +5,15 @@ import React from 'react';
...
  * @param props.message {string}
- * @param props.prompt {prompt}
+ * @param props.askForFeedback {function}
  */
 class ErrorPrompt extends React.Component {
   handleError = ({ error }) => {
-    const { callback, prompt, ...promptProps } = this.props;
-    prompt(callback, promptProps);
+    // Does `promptProps` need to be changed to `rest`?
```
[the remark this answers is not in the document]
Comment (knod) [reply src/components/prompts/ErrorPrompt.js]: Nice, I'll remove this line then.

File: src/components/prompts/ReactRouterConfirmLeave.js (hunk 1 above)
Comment (knod) [reply src/components/prompts/ReactRouterConfirmLeave.js]: I thought this whole thing was triggered when `<VisitPage>` is about to unmount. Isn't that what 'leaving' is in React?

File: src/components/prompts/ErrorPrompt.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (wacii) [reply src/components/prompts/ErrorPrompt.js]: Yeah, I'm just talking about the documentation. `onLeaveCallback` is documented elsewhere and specifying the name here would link to that definition...if we were using jsdoc to process these comments.

File: src/components/prompts/ReactRouterConfirmLeave.js (hunk 2 above)
Comment (knod) [reply src/components/prompts/ReactRouterConfirmLeave.js]: Yeah, I'm still not sure about the definition of prompt. Maybe it's a word we should move away from. Though the folder is called 'prompts'...

Would this work better? 'Whether to interrupt the user's navigation.'

File: src/containers/VisitPage.js (hunk 3 above)
Comment (knod) [reply src/containers/VisitPage.js]: Yeah, I know. I was hoping to save on duplication or on having to make a `resetIfOk` method or some such. Never came to a solid conclusion on that.

File: src/components/prompts/ReactRouterConfirmLeave.js (hunk 1 above)
Comment (wacii) [reply src/components/prompts/ReactRouterConfirmLeave.js]: Ah. I used "leave" as a shortcut for "navigate away from". Just saying `<ReactRouter.Prompt>` would have probably been better in the first place.

File: src/components/prompts/ResetAnytime.js (hunk 4 above)
Comment (knod) [reply src/components/prompts/ResetAnytime.js]: 🎉

File: src/components/prompts/ErrorPrompt.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (knod) [reply src/components/prompts/ErrorPrompt.js]: Oh! I wondered how we could do that with jsdoc. Had no idea that's what it did! That would be awesome! I'll change that back in that case.

File: src/containers/VisitPage.js (hunk 3 above)
Comment (knod) [reply src/containers/VisitPage.js]: Maybe I'll just change the name of that function itself, lol

**Pull request merged**